.github/workflows/main.yml

```
@@ -41,12 +41,45 @@ jobs:
           docker tag ${{secrets.DOCKER_REPO}}:latest ${{secrets.DOCKERHUB_USERNAME}}/${{secrets.DOCKER_REPO}}:latest
           docker push ${{secrets.DOCKERHUB_USERNAME}}/${{secrets.DOCKER_REPO}}
           
-      - name: deploy do-it-client docker image to EC2
+      - uses: actions/checkout@master
+      - name: deploy file transfer by scp
+        uses: siva1024/scp-deployer@latest
+        with:
+          host: ${{ secrets.EC2_HOST }}
+          username: ${{ secrets.EC2_USERNAME }}
+          key: ${{ secrets.EC2_PEM_KEY }}
+          source: "/home/runner/work/do-it-client/do-it-client/deploy/deploy.sh"
```

/home/runner/work/do-it-client/do-it-client/ 이 너무 중복되는것같아서 찾아봤더니
변수로 뺄수있을것같네요
```suggestion
env:
  RUNNER_PATH: /home/runner/work/do-it-client/do-it-client

(중략)
          source: "${{ env.mascot }}/deploy/deploy.sh"
```

https://docs.github.com/en/actions/learn-github-actions/contexts#env-context

---

RUNNER_PATH / AWS_DEPLOY_PATH 변수 추가하여 적용했습니다.